lib/client.js

```diff
@@ -301,23 +299,35 @@ class Client extends EventEmitter {
     return this.request('get', url, data, callback);
   }
 
-  getPendingAuth(url, data, callback, resolve) {
-    if (this.requestsPendingAuth) {
-      this.requestsPendingAuth.push({ url, data, callback, resolve });
-      return;
-    }
+  getPendingAuth(url, data, callback) {
+    return new Promise((resolve, reject) => {
+      if (this.requestsPendingAuth) {
+        this.requestsPendingAuth.push({ url, data, callback, resolve });
+        return;
+      }
 
-    this.requestsPendingAuth = [];
+      this.requestsPendingAuth = [];
 
-    this.authWithKey(() => {
-      this.get(url, data, (_err, result, headers) => {
-        if (callback) {
-          callback(null, result, headers);
-        }
-        resolve(result);
-      });
+      this.authWithKey(() => {
+        this.get(url, data, (err, result, headers) => {
```

Comment:
We need to make sure that the `get` method always returns a `promise`. I think it should be made asynchronous.

In at least one case it will not return a promise - if the `url` is empty.